### PR TITLE
[cloudflare] Fix missing record types

### PR DESCRIPTION
**Pull request by Agreon**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: api.cloudflare.com/#dns-records-for-a-zone-create-dns-record
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/cloudflare/index.d.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ declare namespace Cloudflare {
     }
 
     interface DnsRecordWithPriority {
-        type: Exclude<RecordTypes, 'MX' | 'SRV' | 'URI'>;
+        type: Extract<RecordTypes, 'MX' | 'URI'>;
```
Comment (Xenfo) [inline types/cloudflare/index.d.ts:46]: I believe `SRV` needs to be included:
```
Required for MX, SRV and URI records; unused by other record types. Records with lower priorities are preferred
```

```diff
- type: Extract<RecordTypes, 'MX' | 'URI'>;
+ type: Extract<RecordTypes, 'MX' | 'SRV' | 'URI'>;
```

**Review by samchouse — CHANGES_REQUESTED** [review]

File: types/cloudflare/index.d.ts (hunk 1 above)
Comment (Agreon) [reply types/cloudflare/index.d.ts:46]: If you look at the whole file contents, just below my changes, you will find the Definition of the SRV record, that a previous author added ;)

Comment (Xenfo) [reply types/cloudflare/index.d.ts:46]: Oops, didn’t click in my brain

**Review by samchouse — APPROVED** [review]

**Comment by Agreon** [comment]
Ready to merge

**Pull request merged**